jormungandr-integration-tests/src/common/jormungandr/starter.rs

```diff
@@ -24,235 +24,224 @@ custom_error! {pub StartupError
-    println!("Starting jormungandr node...");
-    let process = command
-        .spawn()
-        .expect("failed to execute 'start jormungandr node'");
-
-    let proces_start_result = process_utils::run_process_until_response_matches(
-        jcli_wrapper::jcli_commands::get_rest_stats_command(&config.get_node_address()),
-        &is_node_up,
-        sleep_between_attempts,
-        max_attempts,
-        "get stats from jormungandr node",
-        "jormungandr node is not up",
-    );
-
-    match proces_start_result {
-        Ok(_) => return Ok(process),
-        Err(e) => {
-            let logger = JormungandrLogger::new(config.log_file_path.clone());
-            logger.print_error_and_invalid_logs();
-            return Err(StartupError::JormungandrNotLaunched { source: e });
-        }
-    }
+#[derive(Clone, Debug, Copy)]
+pub enum StartupVerificationMode {
+    Rest,
+    Log,
 }
 
-fn start_jormungandr_node_sync_with_retry(
-    command: &mut Command,
-    config: &mut JormungandrConfig,
-    timeout: u64,
-    max_attempts: u64,
-) -> JormungandrProcess {
-    let first_attempt =
-        try_to_start_jormungandr_node(command, config.clone(), timeout, max_attempts);
-    match first_attempt {
-        Ok(guard) => return JormungandrProcess::from_config(guard, config.clone()),
-        _ => println!("failed to start jormungandr node. retrying.."),
-    };
-    config.refresh_node_dynamic_params();
-    let second_attempt =
-        try_to_start_jormungandr_node(command, config.clone(), timeout, max_attempts);
-
-    match second_attempt {
-        Ok(guard) => return JormungandrProcess::from_config(guard, config.clone()),
-        Err(e) => {
-            let log_file_content = file_utils::read_file(&config.log_file_path);
-            panic!(format!("{}. Log file: {}", e.to_string(), log_file_content));
-        }
-    };
+#[derive(Clone, Debug, Copy)]
+pub enum OnFail {
+    Retry,
+    Panic,
 }
 
-fn is_node_up(output: Output) -> bool {
-    match output.as_single_node_yaml().get("uptime") {
-        Some(uptime) => {
-            return uptime
-                .parse::<i32>()
-                .expect(&format!("Cannot parse uptime {}", uptime.to_string()))
-                > 2
-        }
-        None => return false,
-    }
+pub trait StartupVerification {
+    fn stop(&self) -> bool;
+    fn success(&self) -> bool;
 }
 
-pub fn start_jormungandr_node(config: &mut JormungandrConfig) -> JormungandrProcess {
-    let mut command = commands::get_start_jormungandr_node_command(
-        &config.node_config_path,
-        &config.genesis_block_path,
-        &config.log_file_path,
-    );
-
-    println!("Starting node with configuration : {:?}", &config);
-    let process = start_jormungandr_node_sync_with_retry(
-        &mut command,
-        config,
-        DEFAULT_SLEEP_BETWEEN_ATTEMPTS,
-        DEFAULT_MAX_ATTEMPTS,
-    );
-    process
-}
+#[derive(Clone, Debug)]
+pub struct RestStartupVerification(JormungandrConfig);
```

Comment:
Unless the struct is an obvious wrapper for the inner member, it's better to have the member named: `self.config` looks much clearer in the method code than `self.0`.